server/assertions/selectors/search.go

```diff
@@ -19,14 +21,23 @@ func filterSpans(rootSpan traces.Span, spanSelector SpanSelector) []traces.Span
 	})
 
 	uniqueSpans := filterDuplicated(filteredSpans)
+	orderedSpans := orderSpansByStartTime(uniqueSpans)
```

Comment:
wouldn't it work better if we ordered the spans when we save them?

Reply:
It's more complex, but yeah, I think it would make it better mainly because we would show the spans in the right order in our DAG view